### PR TITLE
chore: bump konvoy versions in go.mo

### PR DESCRIPTION
**Pull request by dkoshkin**
Bumped the konvoy versions in `go.mod` to match https://github.com/mesosphere/yakcl/blob/cc49d8a53e772f63ce324396fd68ccba0a6fd438/clientapis/go.mod#L30.

The tests passed locally
```
• [SLOW TEST:1779.486 seconds]
[konvoy-e2e] Autoscaling test-suite
/Users/dkoshkin/src/github.com/mesosphere/konvoy/tests/e2e/autoscaling.go:65
  [aws][pull-request] should scale the cluster appropriately regarding its workloads
  /Users/dkoshkin/src/github.com/mesosphere/konvoy/tests/e2e/autoscaling.go:144
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSS
Ran 1 of 31 Specs in 1779.488 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 30 Skipped
```
https://github.com/mesosphere/konvoy/tree/dkoshkin/autoscaling-test

### CODE REVIEW

**Review by jieyu — APPROVED** [review]

File: cluster-autoscaler/go.mod [hunk 1]
```diff
@@ -17,28 +17,27 @@ require (
...
 	gopkg.in/gcfg.v1 v1.2.0
 	gopkg.in/yaml.v2 v2.2.8
-	k8s.io/api v0.17.4
-	k8s.io/apimachinery v0.17.4
+	k8s.io/api v0.18.2
+	k8s.io/apimachinery v0.18.2
 	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
 	k8s.io/cloud-provider v0.16.4
-	k8s.io/component-base v0.17.3
+	k8s.io/component-base v0.17.4
```
Comment (jimmidyson) [inline cluster-autoscaler/go.mod:35]: Why not bump this to be consistent with other `k8s.io/*` dependencies like https://github.com/mesosphere/autoscaler/pull/8/files#diff-5b1211f36242f6afe85bdb0062369dc3R31?

**Review by jimmidyson — APPROVED** [review]
LGTM but a question around keeping consistent k8s.io/* dependency versions

**Comment by hectorj2f** [comment]
I am gonna merge this PR to release a cluster-autoscaler version.

File: cluster-autoscaler/go.mod (hunk 1 above)
Comment (hectorj2f) [reply cluster-autoscaler/go.mod:35]: @dkoshkin please address the comments if needed in a different PR.

File: cluster-autoscaler/go.mod [hunk 2]
```diff
@@ -17,28 +17,27 @@ require (
 	github.com/google/go-querystring v1.0.0
 	github.com/jmespath/go-jmespath v0.0.0-20180206201540-c2b33e8439af
 	github.com/json-iterator/go v1.1.9
-	github.com/mesosphere/kommander-cluster-lifecycle/clientapis v0.0.0-20200323150606-ab6f57ecf9a3
+	github.com/mesosphere/kommander-cluster-lifecycle/clientapis v0.0.0-20200415132633-8e195ef03be9
```
Comment (hectorj2f) [inline cluster-autoscaler/go.mod:20]: this was wrong, we need to use yakcl :/

Comment (hectorj2f) [reply cluster-autoscaler/go.mod:20]: I promise to buy new glasses after this release.

Comment (hectorj2f) [reply cluster-autoscaler/go.mod:20]: I am wondering if this even worked 🤔  ...

File: cluster-autoscaler/go.mod (hunk 1 above)
Comment (dkoshkin) [reply cluster-autoscaler/go.mod:35]: I just copied the new versions and ran `go mod tidy` ,  some transient deps bumped these versions.

**Pull request merged**